crates/control/src/visual_referee_filter.rs

```diff
@@ -0,0 +1,85 @@
+use num_traits::cast::FromPrimitive;
+use rand::prelude::*;
+use std::time::Duration;
+
+use color_eyre::{eyre::Context, Result};
+use context_attribute::context;
+use framework::MainOutput;
+use spl_network_messages::VisualRefereeDecision;
+use spl_network_messages::{PlayerNumber, VisualRefereeMessage};
+use types::{
+    hardware::Interface, messages::OutgoingMessage, CycleTime, FilteredWhistle, PrimaryState,
+};
+
+pub struct VisualRefereeFilter {
+    last_primary_state: PrimaryState,
+}
+
+#[context]
+pub struct CreationContext {
+    pub player_number: Parameter<PlayerNumber, "player_number">,
+}
+
+#[context]
+pub struct CycleContext {
+    pub primary_state: Input<PrimaryState, "primary_state">,
+    pub cycle_time: Input<CycleTime, "cycle_time">,
+    pub filtered_whistle: Input<FilteredWhistle, "filtered_whistle">,
+    pub player_number: Parameter<PlayerNumber, "player_number">,
+
+    pub hardware: HardwareInterface,
+}
+
+#[context]
+#[derive(Default)]
+pub struct MainOutputs {
+    pub last_primary_state: MainOutput<Option<PrimaryState>>,
```

Comment:
Remove? This node should not have any output.

Reply:
Okay.  I thought I must have this.

Reply:
Please keep the empty `MainOutputs` struct, only remove the field from it.

Reply:
Done